packages/babel-plugin-transform-react-display-name/src/index.js

```diff
@@ -52,59 +52,162 @@ export default declare(api => {
     return true;
   }
 
+  function isReactLazy(node) {
+    if (!node || !t.isCallExpression(node)) return false;
+
+    if (!t.matchesPattern(node.callee, "React.lazy")) {
+      return false;
+    }
+
+    return true;
+  }
+
+  function displayNameFromFilename(filename) {
+    if (!filename) {
+      return null;
+    }
+
+    let displayName = path.basename(filename, path.extname(filename));
+
+    // ./{module name}/index.js
+    if (displayName === "index") {
+      displayName = path.basename(path.dirname(filename));
+    }
+
+    return displayName;
+  }
+
+  function displayNameFromAncestors(path) {
+    let id;
+
+    // crawl up the ancestry looking for possible candidates for displayName inference
+    path.find(function(path) {
+      if (path.isAssignmentExpression()) {
+        id = path.node.left;
+      } else if (path.isObjectProperty()) {
```

Comment:
Consider filter out the computed property on path, i.e.
```js
({
    [locale]: React.lazy(() => import("./someComponent"))
});
```
We would not expect `locale` to be `someComponent`'s display name.
```suggestion
      } else if (path.isObjectProperty({ computed: false })) {
```

Reply:
I can see the case for treating this specially for `lazy`, because we commonly have the additional fallback to the `import` path which is probably better than using a computed property key. But we also might not have an `import` - in which case I'd still want to grab a computed key as a last resort.

In any case I will not change the behaviour for `createClass` here since that is out of this PR's scope.

Reply:
> there might not be ... an import path we can use instead ... in which case I'd still want to grab a computed key as a last resort.

True. Actually computed property name will more likely to be combined with template element in import call.

```js
({
    [locale]: React.lazy(() => import(`./${locale}/adsComponent`))
});
```

In this case we can not get a valid import filename. Could you add a test case on the computed property? So we make it clear that it is part of design.